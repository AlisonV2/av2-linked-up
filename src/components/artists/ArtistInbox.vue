<template>
  <div class="row artist-inbox">
    <div class="col-12" v-for="message in messages" :key="message.project">
      <router-link
        :to="{ name: 'MessageDetails', params: { id: message.project } }"
      >
        <div class="message-item">Conversation with : {{ message.client }}</div>
      </router-link>
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistInbox
 * @type {Component}
 * @requires Sentry
 */
export default {
  name: 'ArtistInbox',
  data() {
    return {
      messages: [],
    };
  },
  created() {
    this.$store
      .dispatch('getArtistMessages')
      .then(() => {
        this.messages = this.$store.getters.getArtistMessages;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>

<style lang="scss">
.message-item {
  color: $dark;
  border: 2px solid #dedede;
  background-color: #f1f1f1;
  border-radius: 5px;
  padding: 10px;
  margin: 10px 0;
}
</style>
