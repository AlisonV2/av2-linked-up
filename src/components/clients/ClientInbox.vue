<template>
  <div class="row client-inbox">
    <div class="col-12" v-for="message in messages" :key="message.project">
      <router-link
        :to="{ name: 'MessageDetails', params: { id: message.project } }"
      >
        <div class="message-item">Conversation with : {{ message.artist }}</div>
      </router-link>
    </div>
  </div>
</template>

<script>
/**
 * @exports ClientInbox
 * @type {Component}
 */
export default {
  name: 'ClientInbox',
  data() {
    return {
      messages: [],
    };
  },
  created() {
    this.$store.dispatch('getClientMessages').then(() => {
      this.messages = this.$store.getters.getClientMessages;
    });
  },
};
</script>

<style lang="scss">
.message-item {
  color: $dark;
  border: 2px solid #dedede;
  background-color: #f1f1f1;
  border-radius: 5px;
  padding: 10px;
  margin: 10px 0;
}
</style>
