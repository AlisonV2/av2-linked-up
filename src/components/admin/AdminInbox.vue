<template>
  <ArtistInbox v-if="role === 'artist'" />
  <ClientInbox v-if="role === 'client'" />
</template>

<script>
import * as Sentry from '@sentry/vue';
import ArtistInbox from '@/components/artists/ArtistInbox.vue';
import ClientInbox from '@/components/clients/ClientInbox.vue';

/**
 * @exports AdminInbox
 * @type {Component}
 * @requires Sentry
 * @vue-data{string} role
 * @vue-event {number} getUserRole on created hook
 */
export default {
  name: 'AdminProfile',
  components: {
    ArtistInbox,
    ClientInbox,
  },
  data() {
    return {
      role: '',
    };
  },
  created() {
    this.$store
      .dispatch('getUserRole')
      .then(() => {
        this.role = this.$store.getters.getUserRole;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
