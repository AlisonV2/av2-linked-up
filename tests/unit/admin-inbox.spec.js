import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import AdminInbox from '@/components/admin/AdminInbox.vue';
import ArtistInbox from '@/components/artists/ArtistInbox.vue';
import ClientInbox from '@/components/clients/ClientInbox.vue';

const actions = {
  getUserRole: jest.fn(),
};

const getters = {
  getUserRole: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return {
      role: 'artist',
    };
  },
  global: {
    plugins: [store],
    components: {
      ArtistInbox,
      ClientInbox,
    },
  },
};

/**
 * @module AdminInboxTest
 */
describe('AdminInbox.vue', () => {
  it('AdminInbox Snapshot', () => {
    const component = shallowMount(AdminInbox, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getUserRole action should have been called', async () => {
    const component = shallowMount(AdminInbox, config);
    AdminInbox.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getUserRole).toHaveBeenCalled;
  });

  it('getUserRole getter should have been called', async () => {
    const component = shallowMount(AdminInbox, config);
    AdminInbox.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getUserRole).toHaveBeenCalled;
  });

  it('ArtistInbox should exist', () => {
    const component = shallowMount(AdminInbox, config);
    const item = component.findComponent(ArtistInbox);
    expect(item.exists()).toBe(true);
  });

  it('ClientInbox should exist', () => {
    const component = shallowMount(AdminInbox, {
      ...config,
      data() {
        return {
          role: 'client',
        };
      },
    });
    const item = component.findComponent(ClientInbox);
    expect(item.exists()).toBe(true);
  });
});
