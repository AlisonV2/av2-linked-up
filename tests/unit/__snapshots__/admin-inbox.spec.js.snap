// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminInbox.vue AdminInbox Snapshot 1`] = `
<div
  data-v-app=""
>
  
  <artist-inbox-stub />
  <!--v-if-->
  
</div>
`;
